### DISCUSSION TITLE
What names of .env files does SvelteKit support?

### DISCUSSION THREAD

**Discussion by vekunz** [sveltejs/kit#7383] [Q&A] [answered]
The [documentation](https://kit.svelte.dev/docs/configuration#env) of SvelteKit for the `env` key in the configuration says this to `.env` files:

```
dir — the directory to search for .env files.
```

It clearly says "files" in the plural form. But which files does this mean, since there cannot be multiple files with the same name in one folder? In our projects, we have a `.env` file and locally a `.env.local` file also. Does "files" mean that SvelteKit reads the `.env.local` file also? And if yes, which file overrides which file, if both files contain the same keys?

**Comment by j4w8n** [comment, marked as answer]
SvelteKit uses Vite's `loadEnv()` to load the files. Per the function [here](https://github.com/vitejs/vite/blob/main/packages/vite/src/node/env.ts#L18-L25), it will read the following:

```js
/** default file */ `.env`,
/** local file */ `.env.local`,
/** mode file */ `.env.${mode}`,
/** mode local file */ `.env.${mode}.local`
```

**Reply by j4w8n** [reply]
Based on the code, it processes the files in the above order. Therefore, for example, the value of key `API_KEY` in `.env.local` would override the same key in `.env`.

I have not tested this.
